Userland/Libraries/LibJS/Heap/Cell.h

```diff
@@ -73,6 +74,63 @@ class Cell {
             visit_impl(const_cast<RemoveConst<T>&>(*cell.ptr()));
         }
 
+        template<typename T>
+        void visit(ReadonlySpan<T> span)
+        {
+            for (auto& value : span)
+                visit(value);
+        }
+
+        template<typename T>
+        void visit(Span<T> span)
+        {
+            for (auto& value : span)
+                visit(value);
+        }
+
+        template<typename T>
+        void visit(Vector<T> const& vector)
+        {
+            for (auto& value : vector)
+                visit(value);
+        }
+
+        template<typename T>
+        void visit(HashTable<T> const& table)
+        {
+            for (auto& value : table)
+                visit(value);
+        }
+
+        template<typename T>
+        void visit(OrderedHashTable<T> const& table)
+        {
+            for (auto& value : table)
+                visit(value);
+        }
+
+        template<typename K, typename V, typename T>
+        void visit(HashMap<K, V, T> const& map)
+        {
+            for (auto& [key, value] : map) {
+                if constexpr (IsBaseOf<Cell, K>)
+                    visit(key);
+                if constexpr (IsBaseOf<Cell, V>)
+                    visit(value);
+            }
+        }
+
+        template<typename K, typename V, typename T>
+        void visit(OrderedHashMap<K, V, T> const& map)
+        {
+            for (auto& [key, value] : map) {
+                if constexpr (IsBaseOf<Cell, K>)
+                    visit(key);
+                if constexpr (IsBaseOf<Cell, V>)
+                    visit(value);
```

Comment:
Does this work for e.g. `HashMap<int, GCPtr<something>>` or `HashMap<int, Value>`? In those cases, I don't think we'll pass this check.

Maybe we can get away with a requires clause, something like (untested):
```c++
if constexpr (requires { visit(value); })
    visit(value);
````

Reply:
Yeah this is indeed busted the way I did it. Let's do a `requires` :)